### ISSUE TITLE
Freeze and crash while compiling shaders on Windows 10

### ISSUE THREAD

**Issue by clayjohn** [godlikepanos/anki-3d-engine#75]
When attempting to run the sponza.exe demo or the simple_scene.exe demo I get the following printout and then the full screen goes black (I assume that this is a full screen black window). If I click anywhere the screen goes white and I get the windows crash handler popup. 

I tried letting it run for 10 minutes without touching it in the hopes that it was just a case of Windows freaking out from a slow shader compile, but it appears to be an actual issue. 

I have up to date drivers and have no problems running other Vulkan apps.

I would appreciate any help you can offer. I am hoping to play around with Anki and learn about some of the more advanced graphics features that you have implemented. 

Thanks in advance. 

#### Device info
Device: AMD Radeon RX 5600 XT
Vulkan SDK: first tried with 1.2, then tried with 1.1.130. Both have the same result

#### Command line output
```
[I][CORE] Using settings dir "C:\Users\Clay/.anki" (C:\Users\Clay\dev\anki\anki\core\App.cpp:499 initDirs)
[I][CORE] Initializing application (version 0.1, optimized, NO dbg symbols, NO extra checks, compiler MSVC 192628806, build date Dec 28 2020, commit 9412f12a3) (C:\Users\Clay\dev\anki\anki\core\App.cpp:341 initInternal)
[I][CORE] Number of main threads: 6 (C:\Users\Clay\dev\anki\anki\core\App.cpp:354 initInternal)
[I][CORE] Creating SDL window. SDL version 2.0 (C:\Users\Clay\dev\anki\anki\core\NativeWindowSdl.cpp:39 init)
[I][CORE] SDL window created (C:\Users\Clay\dev\anki\anki\core\NativeWindowSdl.cpp:105 init)
[I][VK  ] Initializing Vulkan backend (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:104 initInternal)
[I][VK  ] Found the following instance extensions: (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:277 initInstance)
[I][VK  ]       VK_KHR_device_group_creation (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_KHR_external_fence_capabilities (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_KHR_external_memory_capabilities (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_KHR_external_semaphore_capabilities (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_KHR_get_physical_device_properties2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_KHR_get_surface_capabilities2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_KHR_surface (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_KHR_win32_surface (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_EXT_debug_report (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_EXT_debug_utils (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ]       VK_EXT_swapchain_colorspace (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:280 initInstance)
[I][VK  ] Will enable the following instance extensions: (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:321 initInstance)
[I][VK  ]       VK_KHR_surface (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:324 initInstance)
[I][VK  ]       VK_KHR_win32_surface (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:324 initInstance)
[I][VK  ]       VK_EXT_debug_report (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:324 initInstance)
[I][VK  ] Number of physical devices: 1 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:373 initInstance)
[I][VK  ] GPU is AMD Radeon RX 5600 XT. Vendor identified as AMD (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:411 initInstance)
[I][VK  ] Number of queue families: 3 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:437 initDevice)
[I][VK  ] Found the following device extensions: (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:494 initDevice)
[I][VK  ]       VK_KHR_16bit_storage (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_8bit_storage (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_bind_memory2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_buffer_device_address (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_create_renderpass2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_dedicated_allocation (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_depth_stencil_resolve (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_descriptor_update_template (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_device_group (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_draw_indirect_count (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_driver_properties (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_external_fence (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_external_fence_win32 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_external_memory (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_external_memory_win32 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_external_semaphore (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_external_semaphore_win32 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_get_memory_requirements2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_imageless_framebuffer (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_image_format_list (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_maintenance1 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_maintenance2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_maintenance3 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_multiview (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_pipeline_executable_properties (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_relaxed_block_layout (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_sampler_mirror_clamp_to_edge (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_separate_depth_stencil_layouts (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_shader_atomic_int64 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_shader_clock (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_shader_draw_parameters (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_shader_float16_int8 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_shader_float_controls (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_shader_non_semantic_info (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_shader_subgroup_extended_types (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_shader_terminate_invocation (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_spirv_1_4 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_storage_buffer_storage_class (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_swapchain (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_swapchain_mutable_format (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_timeline_semaphore (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_uniform_buffer_standard_layout (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_variable_pointers (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_vulkan_memory_model (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_KHR_win32_keyed_mutex (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_calibrated_timestamps (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_conservative_rasterization (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_depth_clip_enable (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_depth_range_unrestricted (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_descriptor_indexing (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_extended_dynamic_state (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_external_memory_host (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_full_screen_exclusive (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_global_priority (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_hdr_metadata (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_host_query_reset (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_image_robustness (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_inline_uniform_block (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_line_rasterization (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_memory_budget (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_memory_priority (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_pipeline_creation_cache_control (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_pipeline_creation_feedback (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_post_depth_coverage (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_private_data (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_queue_family_foreign (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_robustness2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_sampler_filter_minmax (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_sample_locations (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_scalar_block_layout (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_separate_stencil_usage (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_shader_demote_to_helper_invocation (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_shader_image_atomic_int64 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_shader_stencil_export (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_shader_subgroup_ballot (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_shader_subgroup_vote (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_shader_viewport_index_layer (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_subgroup_size_control (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_texel_buffer_alignment (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_tooling_info (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_transform_feedback (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_EXT_vertex_attribute_divisor (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_buffer_marker (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_calibrated_timestamps (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_device_coherent_memory (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_display_native_hdr (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_draw_indirect_count (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_gcn_shader (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_gpa_interface (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_gpu_shader_half_float (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_gpu_shader_int16 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_memory_overallocation_behavior (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_mixed_attachment_samples (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_pipeline_compiler_control (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_shader_ballot (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_shader_core_properties (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_shader_core_properties2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_shader_explicit_vertex_parameter (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_shader_fragment_mask (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_shader_image_load_store_lod (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_shader_info (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_shader_trinary_minmax (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_texture_gather_bias_lod (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_AMD_wave_limits (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_GOOGLE_decorate_string (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_GOOGLE_hlsl_functionality1 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ]       VK_GOOGLE_user_type (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:497 initDevice)
[I][VK  ] Will enable the following device extensions: (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:543 initDevice)
[I][VK  ]       VK_KHR_swapchain (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:546 initDevice)
[I][VK  ] Robust buffer access is disabled (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:561 initDevice)
[I][VK  ] Created a swapchain. Image count: 3, present mode: 0, size: 1920x1080, vsync: 0 (C:\Users\Clay\dev\anki\anki\gr\vulkan\SwapchainFactory.cpp:173 initInternal)
[I][VK  ] Pipeline cache dump not found: C:\Users\Clay/.anki/cache/vk_pipeline_cache (C:\Users\Clay\dev\anki\anki\gr\vulkan\PipelineCache.cpp:55 init)
[I][VK  ] Vulkan memory info: (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:700 initMemory)
[I][VK  ]       Heap 0 size 6157238272 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:703 initMemory)
[I][VK  ]       Heap 1 size 8282701824 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:703 initMemory)
[I][VK  ]       Heap 2 size 268435456 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:703 initMemory)
[I][VK  ]       Mem type 0 points to heap 0, flags 00000000000000000000000000000001 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:709 initMemory)
[I][VK  ]       Mem type 1 points to heap 1, flags 00000000000000000000000000000110 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:709 initMemory)
[I][VK  ]       Mem type 2 points to heap 2, flags 00000000000000000000000000000111 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:709 initMemory)
[I][VK  ]       Mem type 3 points to heap 1, flags 00000000000000000000000000001110 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:709 initMemory)
[I][VK  ]       Mem type 4 points to heap 0, flags 00000000000000000000000011000001 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:709 initMemory)
[I][VK  ]       Mem type 5 points to heap 1, flags 00000000000000000000000011000110 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:709 initMemory)
[I][VK  ]       Mem type 6 points to heap 2, flags 00000000000000000000000011000111 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:709 initMemory)
[I][VK  ]       Mem type 7 points to heap 1, flags 00000000000000000000000011001110 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:709 initMemory)
[I][VK  ] Initializing memory manager (C:\Users\Clay\dev\anki\anki\gr\vulkan\GpuMemoryManager.cpp:201 init)
[I][VK  ]       GPU mem class. Chunk size: 16384, slotSize: 256, allocsPerChunk 64 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GpuMemoryManager.cpp:205 init)
[I][VK  ]       GPU mem class. Chunk size: 262144, slotSize: 4096, allocsPerChunk 64 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GpuMemoryManager.cpp:205 init)
[I][VK  ]       GPU mem class. Chunk size: 8388608, slotSize: 131072, allocsPerChunk 64 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GpuMemoryManager.cpp:205 init)
[I][VK  ]       GPU mem class. Chunk size: 67108864, slotSize: 1048576, allocsPerChunk 64 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GpuMemoryManager.cpp:205 init)
[I][VK  ]       GPU mem class. Chunk size: 134217728, slotSize: 16777216, allocsPerChunk 8 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GpuMemoryManager.cpp:205 init)
[I][VK  ]       GPU mem class. Chunk size: 268435456, slotSize: 67108864, allocsPerChunk 4 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GpuMemoryManager.cpp:205 init)
[I][VK  ]       GPU mem class. Chunk size: 268435456, slotSize: 134217728, allocsPerChunk 2 (C:\Users\Clay\dev\anki\anki\gr\vulkan\GpuMemoryManager.cpp:205 init)
[I][VK  ] R8G8B8 Images are not supported. Will workaround this (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:142 initInternal)
[I][VK  ] S8 Images are supported (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:168 initInternal)
[I][VK  ] D24S8 Images are not supported. Will workaround this (C:\Users\Clay\dev\anki\anki\gr\vulkan\GrManagerImpl.cpp:182 initInternal)
[I][RSRC] Added new data path "C:/Users/Clay/dev/anki" that contains 20300 files (C:\Users\Clay\dev\anki\anki\resource\ResourceFilesystem.cpp:354 addNewPath)
[I][RSRC] Added new data path "C:/Users/Clay/dev/anki/samples/sponza" that contains 256 files (C:\Users\Clay\dev\anki\anki\resource\ResourceFilesystem.cpp:354 addNewPath)
[I][RSRC] Will use 384MB of memory for transfer scratch (C:\Users\Clay\dev\anki\anki\resource\TransferGpuAllocator.cpp:105 init)
[I][RSRC] Compiling shader programs (C:\Users\Clay\dev\anki\anki\resource\ShaderProgramResourceSystem.cpp:58 compileAllShaders)
[I][RSRC]       anki/shaders/GBufferGeneric.ankiprog (C:\Users\Clay\dev\anki\anki\resource\ShaderProgramResourceSystem.cpp:236 skipCompilation)
```

**Comment by godlikepanos** [comment]
Hi and thanks for the report.

10' with 6 CPU cores sounds quite a lot indeed. If you look at the task manager are there any CPU cores working? The compilation should be fully parallel. I suspect there is a bug in glslang or SPIRV tools with multithreaded compilation. As a workaround you can change the `ThreadHive threadHive(getCpuCoresCount(), alloc, false);` to `ThreadHive threadHive(1, alloc, false);` in the `ShaderProgramResourceSystem.cpp`. That may take a lot of time to compile all the shaders though.

**Comment by clayjohn** [comment]
> If you look at the task manager are there any CPU cores working?

Yes, all cores are maxed out.

> As a workaround you can change the ThreadHive threadHive(getCpuCoresCount(), alloc, false); to ThreadHive threadHive(1, alloc, false); in the ShaderProgramResourceSystem.cpp. That may take a lot of time to compile all the shaders though.

I made the change and ran it again. I left it for 25 minutes this time and still it was just the black screen. This time only one core was active, but it was not maxed out. It had spikes and lulls.  

To be clear, is there a chance that shader compiling could take more than 10 or so minutes? I am happy to let it sit for much longer than that if need be.

**Comment by godlikepanos** [comment]
Thanks for the update! 10 mins is quite a lot but please allow me until Monday to test on Windows system (don't have access to one ATM). It the meantime you could revert the change I mentioned and try again for as long as it takes, as long as all cores are working.

In the long run I have an idea that will speed up compilation for this monster shader (if we assume this is the problem). But that will happen in a few weeks.

**Comment by clayjohn** [comment]
Thanks for the response. After my last message I reverted to some older commits to see if I could pinpoint when it last worked. Unfortunately, many of the commits I tested weren't building on windows and the ones that built suffered from the same issue. 

I am not sure what happened in the meantime, but I rebuilt from scratch today after your comment and was able to get it working after clearing out my anki cache. The shaders built in just a few seconds and the test scenes appear to run fine. The only issue I face now is getting it running in RenderDoc. When the app is run from RenderDOC, it crashes with a ``VK_DEVICE_LOST`` error. 

Thank you for taking the time to help me out here. I have a feeling that this turned out to be some weird Windows 10/ AMD driver combination that worked itself out.
